### PR TITLE
Bootstrap requirement with warnings

### PR DESCRIPTION
**Pull request by SamChou19815**
### Summary 

Setup an example requirement with warnings data. Currently the checker is very loose. We could do better but I decide to do it in a future PR, so that this PR is easily reviewbale and @einc can start adding more data without waiting on my work.

### Test Plan 

No impact on other parts of the app (besides the technical elective one) since the new field is just ignored.

### CODE REVIEW

File: src/requirement-types.d.ts [hunk 1]
```diff
@@ -7,6 +7,11 @@ type RequirementCommon = {
   readonly source: string;
   /** If this is set to true, then an edge to the course doesn't count towards double counting. */
   readonly allowCourseDoubleCounting?: true;
```
Comment (einc) [inline src/requirement-types.d.ts:9]: unrelated to your PR, but I just noticed this. why is the type of allowCourseDoubleCounting true?

**Review by einc — APPROVED** [review]
Looks good, thank you Sam!

**Review by tcho6319 — APPROVED** [review]
LGTM

File: src/requirement-types.d.ts (hunk 1 above)
Comment (SamChou19815) [reply src/requirement-types.d.ts:9]: It's an optional field. (See the ?), so when it doesn't exist it just means false

Comment (einc) [reply src/requirement-types.d.ts:9]: ah got it, ty!!

**Pull request merged**